File: problems/arrays-hashing/contains-duplicate/solution.test.ts
import { describe, test, expect } from 'vitest';
import { solutionFunction } from './solution';

describe('Contains Duplicate', () => {
  test('Example 1', () => {
    const input = [1, 2, 3, 1];
    expect(solutionFunction(input)).toBe(true);
  });

  test('Example 2', () => {
    const input = [1, 2, 3, 4];
    expect(solutionFunction(input)).toBe(false);
  });

  test('Edge case: empty input', () => {
    const input: number[] = [];
    expect(solutionFunction(input)).toBe(false);
  });
});

File: problems/arrays-hashing/contains-duplicate/solution_test.go
package contains_duplicate

import "testing"

func TestSolutionFunction(t *testing.T) {
	tests := []struct {
		name     string
		input    []int
		expected bool
	}{
		{
			name:     "Example 1",
			input:    []int{1, 2, 3, 1},
			expected: true,
		},
		{
			name:     "Example 2",
			input:    []int{1, 2, 3, 4},
			expected: false,
		},
		{
			name:     "Edge case: empty input",
			input:    []int{},
			expected: false,
		},
		{
			name:     "Example 2",
			input:    []int{1, 2, 3, 1},
			expected: true,
		},
		{
			name:     "Edge case: empty input",
			input:    nil,
			expected: false,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := SolutionFunction(tt.input)
			if result != tt.expected {
				t.Errorf("got %v, want %v", result, tt.expected)
			}
		})
	}
}
